File: Baekjoon/Class2/10.2609 최대공약수와 최소공배수.js
// 유클리트 호제법
const inputFile = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = require("fs").readFileSync(inputFile, "utf8").trim();

const gcd = (a, b) => { // greatest common divisor
    while(b)
        [a, b] = [b, a % b];
    return a;
}

const lcm = (a, b) => { // least common multiple
    return a * b / gcd(a, b);
}

[a, b] = input.split(" ").map(Number);
console.log(`${gcd(a, b)}\n${lcm(a, b)}`)

File: Baekjoon/Class2/10.2609 최대공약수와 최소공배수.py
#유클리트 호제법
import sys
input = sys.stdin.readline

def gcd(a, b): #greatest common divisor
    while b != 0:
        a, b = b, a % b
    return a

def lcm(a, b): #least common multiple
    return a * b // gcd(a, b)

def main():
    a, b = map(int, input().split())
    print(f"{gcd(a, b)}\n{lcm(a, b)}")

main()